test/fetch-with-retries.spec.ts

```diff
@@ -250,6 +250,35 @@ describe('fetch-with-retries', async () => {
         equal(nockScope.isDone(), true);
     });
 
+    await test('should return the response if ok after retrying 3 times timeout errors (aborted)', async () => {
+        const nockScope = nock('https://test.com')
```

Comment:
I am thinking in change the test here and instead use nock, create a real server and simulate a real timeout. I do not like to fake errors like this. I think it should be easy to do and provide great value.

Reply:
do you still keep this test for some reason or is pending to be deleted?

Also I would move "should throw the error after retrying network errors" to the real-network spec.